presto-spi/src/main/java/com/facebook/presto/spi/block/Block.java

```diff
@@ -144,6 +145,16 @@ default int compareTo(int leftPosition, int leftOffset, int leftLength, Block ri
     }
 
     /**
+     * {@code consumer} visits each of the internal data container and accepts the size for it.
+     * Returns the remaining retained size that has not been visited.
+     * This method can be helpful in cases such as memory counting for internal data structure.
+     */
+    default long getRetainedSizeInBytesBreakdown(BiConsumer<Object, Integer> consumer)
+    {
+        throw new UnsupportedOperationException(getClass().getName());
+    }
```

Comment:
For the API, why not have the implementations call `consumer.accept(this, INSTANCE_SIZE)`?  Then you don't need the return value either.  Also @martint doesn't like calling this "get"